### ISSUE TITLE
Updating function app system key uses systemKey instead of systemKeys resulting in 404

### ISSUE THREAD

**Issue by scp-mb** [Azure/azure-cli#24506]
## Describe the bug

Updating function systemKeys fails because cli expects `systemKey` but API expects `systemKeys` for api version 2022-03-01.

Using the `Try it` functionality for the API docs and passing `systemKeys` works as expected.
https://learn.microsoft.com/en-us/rest/api/appservice/web-apps/create-or-update-host-secret

**Command Name**
`az functionapp keys set`

**Errors:**
```
Operation returned an invalid status 'Not Found'

cli.azure.cli.core.azclierror: az functionapp keys set: 'systemKeys' is not a valid value for '--key-type'. Allowed values: systemKey, functionKeys, masterKey.
```

## To Reproduce:

- `az functionapp keys set -g {} -n {} --key-type systemKey --key-name {} --key-value {}`

## Expected Behavior

--key-type should accept `systemKeys` to match expected API format.

## Environment Summary
```
Windows-10-10.0.19044-SP0
Python 3.10.5
Installer: MSI

azure-cli 2.41.0

Extensions:
azure-devops 0.18.0
dev-spaces 1.0.6

Dependencies:
msal 1.20.0b1
azure-mgmt-resource 21.1.0b1
```

**Comment by yonzhan** [comment]
route to CXP team

**Comment by RakeshMohanMSFT** [comment]
@scp-mb Thank you for reaching out to us, we are looking into it.

**Comment by RakeshMohanMSFT** [comment]
@scp-mb We see that the REST API seems to be accepting SystemKeys and not SystemKey. Let me discuss internally further and update.

**Comment by Francisco-Gamino** [comment]
Adding @amamounelsayed and @kaibocai from the Functions team.

@amamounelsayed / @kaibocai -- Could you please work with the Az CLI team to update the boot to include you both + the new PM owner and remove me and Stefanus? Thanks.

**Comment by scp-mb** [comment]
Confirming this seems to be fixed as of release 2.43.0. Thanks!
